src/meta-srv/src/election/postgres.rs

```diff
@@ -0,0 +1,488 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::{self, Duration};
+
+use common_meta::kv_backend::postgres::{
+    CAS, POINT_DELETE, POINT_GET, PREFIX_SCAN, PUT_IF_NOT_EXISTS,
+};
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt};
+use tokio::sync::broadcast;
+use tokio_postgres::Client;
+
+use crate::election::{
+    Election, LeaderChangeMessage, CANDIDATES_ROOT, CANDIDATE_LEASE_SECS, ELECTION_KEY,
+};
+use crate::error::{
+    DeserializeFromJsonSnafu, PostgresExecutionSnafu, Result, SerializeToJsonSnafu, UnexpectedSnafu,
+};
+use crate::metasrv::{ElectionRef, LeaderValue, MetasrvNodeInfo};
+
+/// Value with a expire time. The expire time is in seconds since UNIX epoch.
+#[derive(Debug, Serialize, Deserialize, Default)]
+struct ValueWithLease {
+    value: String,
+    expire_time: f64,
+}
+
+/// PostgreSql implementation of Election.
+/// TODO: Currently only support candidate registration. Add election logic.
+pub struct PgElection {
+    leader_value: String,
+    client: Client,
+    is_leader: AtomicBool,
+    infancy: AtomicBool,
+    leader_watcher: broadcast::Sender<LeaderChangeMessage>,
+    store_key_prefix: String,
+    candidate_lease_ttl_secs: u64,
+}
+
+impl PgElection {
+    pub async fn with_pg_client(
+        leader_value: String,
+        client: Client,
+        store_key_prefix: String,
+        candidate_lease_ttl_secs: u64,
+    ) -> Result<ElectionRef> {
+        let (tx, _) = broadcast::channel(100);
+        Ok(Arc::new(Self {
+            leader_value,
+            client,
+            is_leader: AtomicBool::new(false),
+            infancy: AtomicBool::new(true),
```

Comment:
The initial value should probably be false.
```suggestion
            infancy: AtomicBool::new(false),
```

Reply:
Yes `EtcdElection` uses false. But seems we should use true here? Since a newly created election is in infancy from my perspective.